### ISSUE TITLE
Merge postgrest-docs with main repo

### ISSUE THREAD

**Issue by steve-chavez** [PostgREST/postgrest#2814]
## Problem

Keeping postgrest-docs in sync with the main repo is quite painful. Duplicate work is done in tagging a release on main, duplicating the CHANGELOG on docs, etc. This makes us slower overall.

## Solution

Merge postgrest-docs with main repo while preserving history. It's possible as previously discussed on: https://github.com/PostgREST/postgrest/pull/1897#issuecomment-881964560

We still need to figure out the steps for CI.

**Comment by steve-chavez** [comment]
If we only make releases through non-main branches, like we have been doing for some patch/minor versions:

- https://github.com/PostgREST/postgrest/tree/rel-10.1.2
- https://github.com/PostgREST/postgrest/tree/rel-10.2.0
- https://github.com/PostgREST/postgrest/tree/rel-11.2.1

Then we would be compatible with the way we release docs too, which is done exclusively on branches: https://github.com/PostgREST/postgrest-docs/branches

Then our main here would point to the latest docs, which is also done for postgrest-docs.

**Comment by steve-chavez** [comment]
Since CI for docs is fast, we could run it on every PR. 

We would have to drop some old version of the docs though, which don't have equivalent branch releases here.

I think this would be the smoother transition.

**Comment by steve-chavez** [comment]
On  https://github.com/PostgREST/postgrest/pull/1897#issuecomment-881957275

> I think we could still maintain the separate repo for tracking issues, even if we move the docs into the code repo
> Benefits
> Docs-related issues/PRs separated from other ones

We agreed to keep the docs issues on its own repo. I think we should keep that for now.

**Comment by laurenceisla** [comment]
Yes, I think moving them to core is the best if archiving the docs repo gives problems.

> It would make the most sense to move them here, but we can't do that 1-by1...

Someone mentioned that bulk issue migration can be done using GitHub CLI (it doesn't keep the labels though): 

https://github.com/orgs/community/discussions/9695#discussioncomment-5423988

**Comment by steve-chavez** [comment]
> So we need to rename our release branches. The obvious choice is rel-X.Y -> vX.Y.
> maybe we should just use the major only and nothing more?

Agree with both :+1:

**Comment by steve-chavez** [comment]
We'll need a redirect on RTD too. Links like https://postgrest.org/en/v12.0/references/api/aggregate_functions.html are not working. We used those on the release page https://github.com/PostgREST/postgrest/releases/tag/v12.0.0.

**Comment by steve-chavez** [comment]
> I then picked 14 issues that I was able to work on quickly in https://github.com/PostgREST/postgrest/pull/3234. This leaves us with 53 issues, but with 25 issues per page, we'd still talk about more than 2 full pages in the issue tracker after migration.

Awesome work! :fire: 

> I think this could be a good compromise.

@wolfgangwalther Agree. LGTM!
